### ISSUE TITLE
Request: OpenWith++ as a direct file opener

### ISSUE THREAD

**Issue by profucius** [stax76/OpenWithPlusPlus#15]
The idea behind OpenWith++ is fantastic, a context menu for opening files from explorer. I present a feature request here which would take OW++ a step further.

Summary: When opening a file, and in which the windows built-in Open With dialog window is displayed, the OpenWith++ program is an option to be selected. When selected, it will redirect through the app, and pop up with the list of apps that this file extension is related to in the configuration.

Purpose: Sometimes you are not just opening a file from explorer, but from within another app. Example: Using Adobe Lightroom, right click on a photo and open it with another app (for further editing). The default windows open with dialog presents itself here.

Similar example: On Android, an app called "Open With" exists, which is similar to this idea, however it works only with web links. You open a link with the Open With app, and it will present you a list of specific apps you can redirect the opening into. One step further, you can save specific apps as the default for specific domains. This concept could be converted into a file opening feature within your app.

**Comment by stax76** [comment]
I'm not sure I fully understand it yet, maybe a concept I've not seen before. Do you still have a use case for it?

**Comment by profucius** [comment]
Sure, I'll expound upon the example I proposed in my initial post. Let me know if I can explain it any further, I want to be sure I am explaining it effectively.

Let's say I am inside of my favorite Photo Manager, such as Adobe Lightroom, or DigiKam (or any app, really, not just photo managers). And let's say I want to open the current photo that I am looking at within another app. Lightroom and DigiKam have a context menu option called "Open in External Editor" (the name does not matter, the idea being that it opens the file via Windows Explorer).

After doing this, the default Windows "Open With" appears, asking which program to open the photo/file with. Sometimes the app that I want to open with is not on this list, or, perhaps I have the app as a "portable" app and so it is not injected into the registry as an available app to open the file with. Or perhaps Windows does not think this file-type can be opened with said desired application.

### Examples of the above: 
- I want to open a JPG file with my Web Browser.
- I want to open a JPG file with my Email client.
- I want to open a DOC file with an open source editor.
- I want to open a CSV file with a portable csv editor app.
- the list goes on.

### Proposal:
OpenWith++ can be added to the default Windows "Open With" dialog for **all** file-types. When selected, the OpenWith++ will then pop-up with a list of app choices based on the settings saved in the main configuration.

Effectively, this makes OpenWith++ "redirect" the "open with" from any file within any windows app, so that you can open that file within any of the apps listed within the OpenWith++ config.

### Reasons:
- Some apps do not play friendly with allowing you to open files with "any" app of your choosing.
- Some apps are not installed, but portable, and therefore not injected into registry, and so Windows does not see them as available "open with" choices.
- Some file-types are not typically opened with certain apps, and so Windows does not offer them as choices.
- All-inclusion. OpenWith++ is already able to open any file with any app via the context menu, so adding this feature is a logical progression to enableing OpenWith++ to open any file with any app via the default Open With dialog as well.

**Comment by stax76** [comment]
Thanks for the high quality feedback!

I'll get back to you.

**Comment by KallistiMan** [comment]
I love this suggestion but I have an additional twist.

I'd like to be able to have a file type associated with OpenWith++, allowing me to open the file with a default app unless I have for instance Ctrl pressed while OpenWith++ starts. Then I'd like to have it opened with a secondary default app (or have the menu open). For me, having a second default app is probably sufficient for most use cases.

My particular use case is for PDF files. I want them to be opened using either Drawboard or Adobe Reader, depending on what the content is. I.e. if it is say a PDF form, it should open in Adobe but for reference articles I want them opened in Drawboard. As with @profucius this also highly relevant when opening files from other apps, when right click isn't available.

So either have the menu open every time, or bypass the menu by having a modifier key pressed.

**Comment by yashpalgoyal1304** [comment]
> if it is say a PDF form, it should open in

yeah, yeah... i have dreamt about such thing for sooo long. like 2 category of apps: viewers, editors. this is applicable to wide variety of file types: images, videos, txt/md files, pdf docs and so on and so forth.

| file types | viewers | editors |
| -- | -- | -- |
image | Google Picasa (propr.) | Gimp, ShareX, ...
pdf | Sumatra PDF | Foxit (propr.)
txt/md | notepad2-mod | others
video | vlc | handbrake, hitfilm (propr.)

there can be more categories in some case, but yeah, thanks a lot @CartCaved (& @profucius too ofc) for bringing this up and reminding me about this.

**Comment by yashpalgoyal1304** [comment]
maybe have the categories to be customisable while having these 2 categories as suggested ones... i dont know, there is always scope of expansions (:

**Comment by yashpalgoyal1304** [comment]
these groups can be implemented in the openwith context menu by using separators as shown in illustrative mockup below:

![vivaldi_1xAYs0pTYP](https://user-images.githubusercontent.com/19423063/143472808-9e599c3b-0e7b-4098-949b-c8bc391e3d7e.png)

**Comment by profucius** [comment]
> Thanks for the high quality feedback!
> I'll get back to you.

Hi again, I'm curious if you've had time over the past year to consider the feature I proposed originally? Would still really love to have this a feature. Thank you.

**Comment by profucius** [comment]
I would just like to follow up on this ticket. It's been a while since I last posted, but I was thinking about this the other day, and would still like to see if there is anything I can offer to help make this feature a reality? Thanks

**Comment by profucius** [comment]
@stax76 Hi again. I thought about this ticket again today. I'm curious if you would be willing to reconsider this feature request? I see the software has not been updated since 2022, however I think it really has a lot of potential to fill a void that currently exists on the windows platform, one that could be extremely useful for power-users trying to avoid redundant and unnecessary clicks during routine workflows.

Just a brief recap of what I'm asking: To make OW++ be an app that is registered in Windows as an app which can itself open files (e.g. the Open File dialog in my earlier screenshot), which then when selected to open a file, will produce a dialog box of the choices configured within the main OW++ app. This would extend OW++ to not only open apps when right-clicking them, but also when sending a file from one app to another, which in essence is the same concept.

**Comment by stax76** [comment]
I use Linux most of the time, mpv.net might be the only Windows app I still work on, unless there is something I need for myself.
